src/researchhub_comment/related_models/rh_comment_model.py

```diff
@@ -77,23 +77,34 @@ def is_root_comment(self):
     """ --- METHODS --- """
 
     @classmethod
-    def create_from_request(cls, request, rh_thread):
-        request_data = request.data
+    def create_from_data(cls, data, rh_thread):
+        from researchhub_comment.serializers.rh_comment_serializer import (
+            RhCommentSerializer,
+        )
+
         [
             comment_content_src_file,
             comment_content_type,
-        ] = cls.get_comment_src_file_from_request(request)
-        rh_comment = cls.objects.create(
-            thread=rh_thread,
-            parent=request_data.get("comment_parent_id"),
-            comment_content_type=comment_content_type,
+        ] = cls.get_comment_src_file_from_request(data)
+        rh_comment_serializer = RhCommentSerializer(
+            {
+                "created_by": data.user.id,
+                "updated_by": data.user.id,
+                "parent": data.get("comment_parent_id"),
+                "comment_content_type": comment_content_type,
+                "thread": rh_thread,
+            }
         )
-        rh_comment.comment_content_src.save(
-            f"RH-THREAD-{rh_thread.id}-COMMENT-{rh_comment.id}-user-{request.user.id}.txt",
-            comment_content_src_file,
-        )
-        rh_comment.refresh_from_db()
-        return rh_comment
+        try:
+            rh_comment_serializer.is_valid(raise_exception=True)
+            rh_comment = rh_comment_serializer.save()
+            rh_comment.comment_content_src.save(
+                f"RH-THREAD-{rh_thread.id}-COMMENT-{rh_comment.id}-user-{data.user.id}.txt",
+                comment_content_src_file,
+            )
+            return rh_comment
+        except Exception as error:
+            raise Exception(f"Failed to RhCommentModel#create_from_data: {error}")
```

Comment:
We can remove this try catch, since it doesn't really do anything.

Reply:
it does. It will show where the error happened in terms of code location & it will log the serializer validation stuff

Reply:
The view that's calling `create_from_data` already has a try catch

Reply:
Yes, but the point of having a class method is to be able to use it in more than one place. 
We cannot only rely on the call site.

Reply:
But is there a point of having it wrapped in a try block just to raise another exception?

Reply:
Yes, that way we know exactly where the exception was raised. From my experience, django doesn't bubble up the exceptions when it's caught at lower levels like this. This is good because it's won't double log & helps the devs to know exactly where it's throwing.

Reply:
I don't really see how this will help locate the error. In some cases, try catches usually make it harder to find where an exception is thrown (e.g., paper serializer/views ). I don't think this will be the case for this specific try catch, but if you see my example down below, catching the error makes no difference?

Another example:

Without a try/catch
```
  ...
  File "/Users/.../rh_comment_model.py", line 99, in create_from_data
    rh_comment_serializer.is_valid(raise_exception=True)
  ValidationError: 'foo'
```
vs

With a try/catch
```
  ...
  File "/Users/.../rh_comment_model.py", line 107, in create_from_data
    raise Exception(f"Failed to RhCommentModel#create_from_data: {error}")
  File "/Users/.../rh_comment_model.py", line 99, in create_from_data
    rh_comment_serializer.is_valid(raise_exception=True)
  Exception: 'Failed to RhCommentModel#create_from_data:  foo'
```